testing/unit-testing-jest/docs/spy-external.md

```diff
@@ -0,0 +1,132 @@
+# Spy external function
+Component
+````ts
+@Component({
+  selector: 'app-example',
+  templateUrl: `
+    <div class="section" (click)="log('test')">
+        Click me
+    </div>
+  `
+})
+export class ExampleComponent {
+    constructor(private logService: LogService){}
+
+    log(message: string): void {
+        this.logService.log(message); 
+    }
+}
+````
+
+Test
+````ts
+const fakeLogService = {
+    log: jest.fn()
+}
+
+describe('ExampleComponent', () => {
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+          declarations: [ExampleComponent],
+          providers: [
+            {
+                provide: LogService,
+                useValue: fakeLogService
+            }
+          ]
+        });
+    })
+    
+    it('should call log function', () => {
+        fixture.nativeElement.querySelector('.section').click();
+        expect(fakeLogService.log).toHaveBeeCalledWith('test');
+    });
+}
+````
```

Comment:
Add some description explaining what do you do here, what a newbie dev should look at. Guide readers.

Reply:
Also check identation